### DISCUSSION TITLE
New Feature: Web Component

### DISCUSSION THREAD

**Discussion by mar10** [mar10/wunderbaum#33] [Ideas]
Should Wunderbaum register a web component?
E.g.
```html
<wunderbaum-tree
    source='URL'>
</wunderbaum-tree>
```

Some questions come to mind:

- Should this be part of the core or a separate npm package?
  See [this PR](https://github.com/mar10/wunderbaum/pull/28)
  (note that we do not want to introduce external dependencies)
- Does it even make sense?
  - There are currently a lot of options, and not all of them are scalar values that can easily be passed as element attributes.
  - There are also many callbacks available, that we would need to convert to native events?

**Comment by jogibear9988** [comment]
I'll use wunderbaum inside of components for my designer, so I would see what I need. When I finish with them, I create a small (zero dependecy) sample for a usable webcomponent
